Ensure image name is clean

djberg96 opened the pull request:
The `build_image_name` method will create a leading "./" if the dirname is relative. This is because `File.dirname("foo")` returns "." on relative path names.

~~WIP for now until I can see if there's a nicer way without involving Pathname.~~

I didn't see a better way, so Pathname it is.

See also https://github.com/ManageIQ/manageiq-schema/pull/198

djberg96 commented:
@bzwei What do you think? Is there a nicer way?

Also, this will require a migration.

bzwei commented:
@djberg96 what's the problem - is it a bug or simply a presentation issue? Normally "." in a path does not cause a problem, therefore a migration is not needed.

djberg96 commented:
@bzwei Presentation issue. Seems weird to have "./foo" as an image name. I'll double check, but without the migration users would end up with "./foo" and "foo" as separate records, wouldn't they?

djberg96 commented:
@bzwei I added a migration in the schema repo as well.

bzwei commented:
consider to add a spec test?

djberg96 commented:
@bzwei Alright, I guess I added them for the migration but not here. Will add them.

djberg96 commented:
@bzwei Ok, added specs. Also, I moved it into the helper methods module so that I could more easily test it, and possibly re-use it in the targeted refresh code eventually.

bzwei commented on app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb:
can you use `build_image_name` method in `refresh_helper_methods`?

djberg96 replied:
I'd have to include the module. I don't think it will be an issue, but I'll give it a try.
